firmware/ventilator-controller-stm32/Core/Test/Pufferfish/Protocols/messages.txt

```diff
@@ -0,0 +1,207 @@
+
+    WHEN("A body with an encoded payload and 1 byte header of value equal to 0 is parsed")
+      THEN("The parse method reports invalid type status")
+      THEN("After the parse method is called, The value assigned to the type member is equal to the type field of the input_buffer body's header")
+      THEN("The payload Taggedunion tag is set to Unknown")
+      THEN("the payload.values field remains unchanged")
+
+    WHEN("A body with an empty payload and 1 byte header of value equal to the message descriptor array size is parsed")
+      THEN("The parse method reports invalid type status")
+      THEN("After the parse method is called, The value assigned to the type member is equal to the type field of the input_buffer body's header")
+      THEN("The payload Taggedunion tag is set to Unknown")
+      THEN("the payload.values field remains unchanged")
+
+    WHEN("A body with an empty payload and 1 byte header of value greater than the message descriptor array size is parsed")
+      THEN("The parse method reports invalid type status")
+      THEN("After the parse method is called, The value assigned to the type member is equal to the type field of the input_buffer body's header")
+      THEN("The payload Taggedunion tag is set to Unknown")
+      THEN("the payload.values field remains unchanged")
+
+    WHEN("A body with a 1 byte header and payload of sensor_measurements message with invalid encoding is parsed")
+      THEN("The parse method reports invalid encoding status")
+      THEN("After the parse method is called, The value assigned to the type member is equal to the input_buffer body's header")
+      THEN("The payload Taggedunion tag is set to sensor_measurements")
+      THEN("The message payload fields are not set")
+
+/* test cases for parse functions for each message type
+    WHEN("A body with 1 byte header and a paylod of {Message type} message type is parsed")
+      THEN("The parse method reports ok status")
+      THEN("After the parse method is called, The value assigned to the type member is equal to the input_buffer body's header")
+      THEN("The payload Taggedunion tag is set to {Message type}")
+      THEN("The message payload field values are as expected")
+      THEN("The input buffer is unchanged after parse")
+/*
+
+Scenario: The parse method correctly parses bodies when the payload and field members have data
+
+  GIVEN("A Message object constructed with StateSegment Taggedunion, default message types and a max size of 252 bytes")
+  // setup:
+  // type = 5
+  // parameters_request.fio2 = 80;
+  // parameters_request.mode = VentilationMode_hfnc;
+  // parameters_request.ventilating = true;
+
+    WHEN("An empty input buffer body is parsed")
+      THEN("The parse method reports invalid length status")
+      THEN("After the parse method is called, The value assigned to the type member remains unchanged")
+      THEN("The payload Taggedunion tag is set to Unknown")
```

Comment:
Are you sure that the value of the tag is changed to be unknown, instead of being left unchanged?

Reply:
Ya my bad, it is left unchanged.